src/main/java/uk/gov/hmcts/darts/arm/util/ArmResponseFilesUtil.java

```diff
@@ -37,4 +41,28 @@ public static String getObjectTypeId(ExternalObjectDirectoryEntity externalObjec
         }
         return objectTypeId;
     }
+
+    public static Integer getRelationIdFromArmResponseFilesInputField(String input) {
+        try {
+            String relationId = findFirstPatternMatch("\"relation_id\".*?:.*?\"([^\"]+)\"", input);
+            return Integer.parseInt(relationId);
+        } catch (Exception e) {
+            log.error("Error: {}", e.getMessage());
+        }
+        return null;
+    }
```

Comment:
@hemantasharma1129  @jackmaloney  if we get invalid JSON returned to us should we still be trying to to get a relation_id or should we log this for live support to work out?

Conscious this could cause us issues if we grab the incorrect relation_id

Reply:
We must extract the relationId from IL file for error handling. Thats how ARM will respond back so DARTS should be smart enough to extract it and update the DB with correct error message.

Reply:
Does this include when the JSON given to us is invalid?
It feels very risky to try and pull out the value from invalid json? as we could get the incorrect value and process it as if it is valid causing data integrity issues

Reply:
If the question is why reading the JSON, then answer is we must read it to make sure we update the DARTS system in case DARTS sends an INVALID manifest to ARM. Even if its invalid, we should be able to extract the relation_id to update DARTS db.
If the concern is around the Regex solution, happy to hear any other solution which is much safer to allow us to extract the relation_id from the IL file.

Reply:
I think any solution we have if we get invalid json (As in json structure is invalid not just wrong fields) something more fundamental has gone wrong that we need to fix first then we can replay these fields. It feels very risky to try and process invalid json to pull out the values, as we have 0 control or validation against this invalid json response, so it could cause security vulnerabilities as we are accepting what every they give us as well as data integrity issues as we are marking fields as invalid from the invalid json we get but the invalid Json could have returned us the wrong data (Which is likely if it is invalid json in the first place) so we could be marking the incorrect records as invalid. 

I believe the safest approach is to log the invalid json and skip processing then raise an alert on DynaTrace for live support team to investigate and if they want to manually replay or mark the file as failed.

Reply:
I'm happy for all other failures but for invalid json as it is a critical failure I don't think we should try to derive the values but instead force the logging route for it.

But for all other failures we can convert the object into json and grab the value (instead of using regex).

@hemantasharma1129  is this what you mean? If not it maybe worth having a quick 5-10 min chat on this